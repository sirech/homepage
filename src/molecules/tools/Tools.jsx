import React from 'react'

import { Card, CardHeader, CardText, CardBlock, Badge } from 'reactstrap'

const tools = [
  'spacemacs',
  'zsh',
  'git',
  'tmux',
  'Google Chrome',
  'hammerspoon',
  'Alfred'
]

const renderTool = (name) => (
  <Badge key={name} className='mr-1' pill color='default'>{name}</Badge>
)

const renderTools = () => (
  tools.map(name => renderTool(name))
)

const Tools = () => (
  <Card id='tools'>
    <CardHeader tag='h5'>Tools</CardHeader>
    <CardBlock>
      <CardText>
        I am a bit picky about the tools I use. Actually, I am <b>very picky</b>. I cannot remember how many times I did rewrite my shell config in the past. It is problematic for pair programming because I usually am the only one that can work at my computer.
      </CardText>

      <CardText>
        Nowadays I try to at least outsource my configurations a bit, using some awesome community driven ones like <a href='http://spacemacs.org'>Spacemacs</a> and <a href='https://github.com/robbyrussell/oh-my-zsh'>Oh My Zsh</a>
      </CardText>
      {renderTools()}
    </CardBlock>
  </Card>
)

export default Tools
