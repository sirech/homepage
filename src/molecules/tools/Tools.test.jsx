import React from 'react'
import renderer from 'react-test-renderer'
import Tools from './Tools'

describe('components', () => {
  describe('Tools', () => {
    it('renders correctly', () => {
      const component = renderer.create(<Tools />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
