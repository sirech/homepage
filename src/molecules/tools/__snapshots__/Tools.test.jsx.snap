// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Tools renders correctly 1`] = `
<div
  className="card"
  id="tools"
>
  <h5
    className="card-header"
  >
    Tools
  </h5>
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      I am a bit picky about the tools I use. Actually, I am 
      <b>
        very picky
      </b>
      . I cannot remember how many times I did rewrite my shell config in the past. It is problematic for pair programming because I usually am the only one that can work at my computer.
    </p>
    <p
      className="card-text"
    >
      Nowadays I try to at least outsource my configurations a bit, using some awesome community driven ones like 
      <a
        href="http://spacemacs.org"
      >
        Spacemacs
      </a>
       and 
      <a
        href="https://github.com/robbyrussell/oh-my-zsh"
      >
        Oh My Zsh
      </a>
    </p>
    <span
      className="mr-1 badge badge-secondary badge-pill"
    >
      spacemacs
    </span>
    <span
      className="mr-1 badge badge-secondary badge-pill"
    >
      zsh
    </span>
    <span
      className="mr-1 badge badge-secondary badge-pill"
    >
      git
    </span>
    <span
      className="mr-1 badge badge-secondary badge-pill"
    >
      tmux
    </span>
    <span
      className="mr-1 badge badge-secondary badge-pill"
    >
      Google Chrome
    </span>
    <span
      className="mr-1 badge badge-secondary badge-pill"
    >
      hammerspoon
    </span>
    <span
      className="mr-1 badge badge-secondary badge-pill"
    >
      Alfred
    </span>
  </div>
</div>
`;
